src/System.IO.FileSystem/tests/Base/AllGetSetAttributes.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // Tests that are valid for File, FileInfo, and DirectoryInfo
+    public abstract class AllGetSetAttributes : BaseGetSetAttributes
+    {
+        [Fact]
+        public void NullParameters()
+        {
+            Assert.Throws<ArgumentNullException>(() => Get(null));
+            Assert.Throws<ArgumentNullException>(() => Set(null, FileAttributes.Normal));
```

Comment:
I didn't look to add missing parameter name validation (or any other potentially missing tests) in this pass. I moved tests and "commonified" them.

I'm planning to have my interns get their feet wet on some really simple tests- I think looking for missing name validation would be a good first task.